exist-core/src/main/java/org/exist/xquery/ConditionalExpression.java

```diff
@@ -92,8 +92,9 @@ public void analyze(AnalyzeContextInfo contextInfo) throws XPathException {
      */
     public Sequence eval(Sequence contextSequence, Item contextItem) throws XPathException {
         context.expressionStart(this);
-        final Sequence testSeq = testExpr.eval(contextSequence, contextItem);
+
         try {
+            final Sequence testSeq = testExpr.eval(contextSequence, contextItem);
```

Comment:
This assumes that the `testExpr` is on the same line as the conditional expression. The line/column number will be close but not necessarily accurate. I wonder if it is better to have no line number, or to have a line number which is in the vicinity of the error?

Although... I note that this is already problematic below for the `thenExpr` and `elseExpr`.

Reply:
yes i realised after closing the laptop yesterday that the solution is just too simple :-)